boa/src/builtins/value/mod.rs

```diff
@@ -737,6 +754,9 @@ impl ValueData {
                 // TODO: throw TypeError
                 panic!("TypeError: \"BigInt value can't be serialized in JSON\"");
             }
+            Self::Symbol(_) | Self::Undefined => {
+                panic!("Symbols and Undefined JSON Values depend on parent type");
```

Comment:
Just for my understanding, is this in the spec? Should we check this before calling? Or should this throw an exception?

Reply:
@Razican 
> Just for my understanding, is this in the spec? Should we check this before calling?
According to NOTE 5 in the spec, JSON.stringify should return undefined. However, symbols, functions, undefined do not have corresponding counterparts, and thus, it depends on what their parent is. I did everything I could to keep that knowledge in the to_json function.

> Or should this throw an exception?
I am not sure. I can look that up later, unless somebody else reviewing knows that answer. cc: @HalidOdat

Reply:
From my understanding when calling `JSON.stringify` with `symbol` or `undefined` it should return `undefined`. A workaround would be to check if it is `undefined` or `symbol` in `JSON.stringify` and return undefined before calling `Value::to_json`.

What do you think ? @Razican @jasonwilliams

Reply:
@HalidOdat You are definitely correct WRT JSON.stringify returning undefined for symbols, functions and undefined inputs. I added some tests and updated to pass those tests. The question does still remain here on whether or not to panic. My thought process behind panicking was that if this branch is hit, it is because the calling function needs to defend against these values. Thus, when this panics we'll just need to create a bug and address it.

Reply:
> @HalidOdat You are definitely correct WRT `JSON.stringify` returning undefined for symbols, functions and undefined inputs. I added some tests and updated to pass those tests. The question does still remain here on whether or not to panic. My thought process behind panicking was that if this branch is hit, it is because the calling function needs to defend against these values. Thus, when this panics we'll just need to create a bug and address it.

I would use `unreachable!()` for this.

Reply:
:heavy_check_mark: